monitoring/bin/deploy_dashboards.sh

```diff
@@ -25,108 +25,108 @@ TEST_DASH="${TEST_DASH:-false}"
 DASH_BASE="${DASH_BASE:-monitoring/dashboards}"
 
 function deploy_dashboards {
-   type=$1
-   if [ -z "$2" ]; then
-     dir=$"$DASH_BASE/$type"
-   else
-     dir=$2
-   fi
-   
-   for f in $dir/*.json; do
-     # Need to check existence because if there are no matching files,
-     # f will include the wildcard character (*)
-     if [ -f "$f" ]; then
-       log_debug "Deploying dashboard from file [$f]"
-       name=$(basename $f .json)
-       
-       kubectl create cm -n $DASH_NS $name --dry-run=client --from-file $f -o yaml | kubectl apply --server-side -f -
-       kubectl label cm -n $DASH_NS $name --overwrite grafana_dashboard=1 sas.com/monitoring-base=kube-viya-monitoring sas.com/dashboardType=$type
-     fi
-   done
+    type=$1
+    if [ -z "$2" ]; then
+        dir="$DASH_BASE/$type"
```

Comment:
Interesting.  It appears that calling this function with a single argument (which corresponds to a sub-directory under $DASH_BASE _only_ works from _within_ this script.  Calling the script (rather than the function) with a single argument will fail if the argument does not refer to a full directory (due to the validation around line 67).  

I'm not sure that's optimal but I guess this is how things have worked for a very long time, so we leave it as-is.

Reply:
Yea, that's a good point. Think it's worth making a ticket to address this later? Or worth me looking for a solution now?

Reply:
My initial reaction was 'no, I think we just accept it as a "quirk"'.  But then I checked the documentation and we do say it can be used to deploy a single dashboard.  I don't think this is critical, so no need to address it now; just add a story to the backlog.